### PR TITLE
[FLINK-22847][sql-client] Quote options when using SET;

### PR DESCRIPTION
**Pull request by Airblader**
## What is the purpose of the change

Quotes options when executing `SET;` to list all current options. This aligns the output with the new quoted syntax added to the parser. While SQL client does currently still support the unquoted syntax for compatibility, we eventually want to deprecate and remove it as it requires special parsing outside the parser. This change encourages users to use the quoted syntax by consistently quoting in Flink itself.

## Brief change log

* Quote options when calling `SET;`

## Verifying this change

This change is already covered by existing tests.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

**Review by twalthr — APPROVED** [review]
LGTM, btw can we also update the files in `src/test/resources/sql/` for completeness. maybe you can add this as a hotfix to this PR. I will merge it once the build is green.

**Comment by Airblader** [comment]
I've updated those files now as well.

File: flink-table/flink-sql-client/src/test/resources/sql/misc.q [hunk 1]
```diff
@@ -30,9 +30,9 @@ HELP		Prints the available commands.
 INSERT INTO		Inserts the results of a SQL SELECT query into a declared table sink.
 INSERT OVERWRITE		Inserts the results of a SQL SELECT query into a declared table sink and overwrite existing data.
 QUIT		Quits the SQL CLI client.
-RESET		Resets a session configuration property. Syntax: 'RESET <key>;'. Use 'RESET;' for reset all session properties.
+RESET		Resets a session configuration property. Syntax: 'RESET '<key>';'. Use 'RESET;' for reset all session properties.
```
[the remark this answers is not in the document]
Comment (twalthr) [reply flink-table/flink-sql-client/src/test/resources/sql/misc.q:33]: Let's change them to double quotes. It is just explanatory text.

[the remark this answers is not in the document]
Comment (Airblader) [reply flink-table/flink-sql-client/src/test/resources/sql/misc.q]: Willdo. Just noticed now these are actually tests(!) and not just documentation. :-D Had to be a bit more careful here.

**Pull request closed without merging**